backend/src/resolvers.js

```
@@ -0,0 +1,56 @@
+export const resolvers = {
+  Query: {
+    // listallposts
+    posts: async (parent, args, context) => {
+      return context.dataSources.postdb.getAll()
+    },
+    // listallusers
+    users: async (parent, args, context) => {
+      return context.dataSources.userdb.getAll()
+    }
+  },
+  Mutation: {
+    // signup
+    signup: async (parent, args, context) => {
+      return context.dataSources.userdb.signup(args)
+        .catch((err) => {
+          return err
```

Exception was not handled properly. 
``` return '"An error occured while signing up. " + err; ```